lms/views/basic_lti_launch.py

```diff
@@ -36,36 +36,41 @@ class BasicLTILaunchViews:
     def __init__(self, context, request):
         self.context = context
         self.request = request
+        self.lis_result_service = request.find_service(name="lis_result_sourcedid")
 
         # Configure the front-end mini-app to run.
         self.context.js_config.update({"mode": "basic-lti-launch"})
 
-        # Add config used by frontend to call `record_submission` API.
-        params = self.request.params
-        if (
-            # The outcome reporting params are typically only available when
-            # students (not teachers) launch an assignment.
-            params.get("lis_result_sourcedid")
-            and params.get("lis_outcome_service_url")
-            # This feature is initially restricted to Canvas.
-            and self._is_launched_by_canvas()
-        ):
-            self.context.js_config["submissionParams"] = {}
-
-            self._set_submission_param("h_username", context.h_user.username)
-            self._set_submission_param(
-                "lis_result_sourcedid", params.get("lis_result_sourcedid")
-            )
-            self._set_submission_param(
-                "lis_outcome_service_url", params.get("lis_outcome_service_url")
-            )
+        self._add_canvas_submission_params()
 
         # If the launch has been configured to focus on the annotations from
         # a particular user, translate that into Hypothesis client configuration.
         focused_user = self.request.params.get("focused_user")
         if focused_user:
             self._set_focused_user(focused_user)
 
+    def _add_canvas_submission_params(self):
+        """
+        Add config used by frontend to call Canvas `record_submission` API.
+
+        The outcome reporting params are typically only available when
+        students (not teachers) launch an assignment.
+        """
+
+        if not self._is_launched_by_canvas():
+            return
+
+        lis_params = self.lis_result_service.get_grading_parameters(self.request)
+
+        if not lis_params:
+            return
```

Comment:
I guess this will now return if `context_id` and/or `resource_link_id` is missing (whereas previously the `js_config` below would have been set as long as `lis_result_sourcedid` and `lis_outcome_service_url` were present). In practice I don't think this makes a difference -- `context_id` and `resource_link_id` _are_ present in Canvas launches.